sktime/datatypes/_dtypekind.py

```diff
@@ -76,6 +78,38 @@ def _pandas_dtype_to_kind(col_dtypes):
         )
 
 
+if _check_soft_dependencies(["polars", "pyarrow"], severity="none"):
+
+    def _polars_dtype_to_kind(col_dtypes):
+        """Convert polars dtypes to enum DtypeKind values."""
+        import polars as pl
+
+        if isinstance(col_dtypes, list):
+            return [_polars_dtype_to_kind(x) for x in col_dtypes]
+
+        if col_dtypes.is_float():
+            return DtypeKind.FLOAT
+        elif col_dtypes.is_signed_integer():
+            return DtypeKind.INT
+        elif col_dtypes.is_unsigned_integer():
+            return DtypeKind.UINT
+        elif isinstance(col_dtypes, pl.Categorical) or isinstance(
```

Comment:
Everything else looks good to me. I think this can instead be written as isinstance(col_dtypes, (pl.Categorical,pl.Object))